lib/engine/game/g_18_usa/game.rb

```diff
@@ -1537,30 +1642,136 @@ def randomize_subsidies
           end
         end
 
+        def home_hex_for(corporation)
+          corporation.tokens.first.hex
+        end
+
+        #
+        # In 18USA you need to use the maximum number of exits for a given tile, but unlike 1817 there are more types of tiles
+        # that this applies to:
+        # Gray plain track
+        # Brown cities
+        # Gray plain track with labels
+        # and at a given time it's possible to have multiple legal color choices to lay
+        # so we need to be able to filter within each group
+        def filter_by_max_edges(tiles)
+          tiles.group_by { |t| [t.color, t.cities&.size, t.label&.to_s] }.flat_map do |grouped_tiles|
+            #flat_map on the hash flattens the hash into [[key, value], [key value], [key, value], ...]
+            max_edges = grouped_tiles.last.map { |t| t.edges.length }.max
```

Comment:
edges.size

Reply:
grouped_tiles = grouped_tiles.last

.grouped_tiles.map ...
group_tiles.select

Reply:
The usage of `edges.length` is replaced with `edges.size`
I fixed the rubocop violation I introduced earlier today (no space after #)
I reassign `grouped_tiles.last` to `grouped_tiles` as you mentioned